frontend/server/src/Controllers/Contest.php

```diff
@@ -2671,6 +2676,19 @@ public static function apiCreate(\OmegaUp\Request $r) {
             ['partial','all_or_nothing','max_per_group'],
         );
         $checkPlagiarism = $r->ensureOptionalBool('check_plagiarism') ?? false;
+        
+        // Handle recommended flag - only available for admins and support team
+        $recommended = false;
+        if ($r->ensureOptionalBool('recommended') ?? false) {
+            if (!\OmegaUp\Authorization::isSystemAdmin($r->identity) && 
+                !\OmegaUp\Authorization::isSupportTeamMember($r->identity)) {
+                throw new \OmegaUp\Exceptions\ForbiddenAccessException(
+                    'userNotAllowed'
+                );
+            }
+            $recommended = true;
+        }
```

Comment:
What do you think if we only set the value of `recommended` when a user has a valid role?

Since the UI shouldn't display this field for regular users, we should only omit the value if someone calls the API from outside the interface. Something like:
```suggestion
        if (\OmegaUp\Authorization::isSupportTeamMember($r->identity)) {
            $recommended = $r->ensureOptionalBool('recommended') ?? false;
        }
```
By the way, you'll need to add `recommended` as a `request-param` at the beginning of the function.